napari/layers/image/image.py

```diff
@@ -917,7 +917,12 @@ def _update_thumbnail(self):
             image = np.max(image, axis=0)
 
         # float16 not supported by ndi.zoom
-        dtype = np.dtype(image.dtype)
+        try:
+            dtype = np.dtype(image.dtype)
+        except TypeError:
+            # tensorstore case
+            dtype = np.dtype(image.dtype.type)
+
```

Comment:
Can we be more explicit here and have an if isinstance(...) or something else ?

Reply:
I do not know how. Problem is with tensorstore dtype, and tensorstore is optional depenecy.

Reply:
Hum, can we create an issue once this is merge to track being explicit ? 
Something like `image.dtype.__class__.__name__.startswith('tensorstore.')` ?

Reply:
I do not observe this problem on the main branch. It happens only on the async test on 0.4.19 branch.

Reply:
Oh, this target 4.19 ! Sorry, then nevermind, this is all good.